mpire/signal.py

```diff
@@ -32,13 +29,15 @@ def __exit__(self, exc_type: Type, exc_val: Exception, exc_tb: Traceback) -> Non
 class DisableKeyboardInterruptSignal:
 
     def __enter__(self) -> None:
-        # Prevent signal from propagating to child process
-        self._handler = getsignal(SIGINT)
-        ignore_keyboard_interrupt()
+        if current_thread() == main_thread():
```

Comment:
Does this mean that on the main thread keyboard signals are disabled? Then can you cntrl + c when a WorkerPool is running on the main thread for example?

Reply:
These signal handling classes temporarily disable or postpone signals. This is useful when mpire starts the child processes. When you disable signals when you create child processes, those child processed won't be killed when you use ctrl + c. The main process still receives the signal though. This is good because I can then gracefully shutdown the workers, instead of having deadlocks.

And I usually use `DelayedKeyboardInterrupt` because then the interrupt signal is blocked inside the with clause, but once it exits it sends out any interrupt signal that was send during that period, so it still gets processed. 

I hope this makes sense, otherwise ask me again.